detectable/src/main/java/com/synopsys/integration/detectable/detectables/gradle/inspection/parse/GradleReportLineParser.java

```diff
@@ -134,13 +136,16 @@ private void addRelation(String projectParent, String projectName, String rootPr
     }
 
     private boolean checkParentRichVersion(String rootProjectName, String projectParent, String dependencyGroupName) {
-       String currentProject = projectParent.substring(projectParent.lastIndexOf(":") + 1, projectParent.lastIndexOf("'"));
-       while(!currentProject.equals(rootProjectName)) {
-           if(richVersionGroup.containsKey(currentProject) && richVersionGroup.get(currentProject).contains(dependencyGroupName)) {
-               return true;
-           }
-           currentProject = relationsMap.getOrDefault(currentProject, rootProjectName);
-       }
+        if(!projectParent.equals("null")) {
+            String currentProject = projectParent.substring(projectParent.lastIndexOf(":") + 1, projectParent.lastIndexOf("'"));
```

Comment:
Should we handle '-1' / out of bounds here?

Reply:
Yeah good thought, I will add a check there. Thanks